kore/src/Kore/Step/Simplification/Overloading.hs

```diff
@@ -239,8 +295,160 @@ unifyOverloadingVsOverloaded
         then return $ Pair overloadingTerm overloadedTerm'
         else throwBottom "different injected ctor"
 
+{- Handles the case
+    overloadingTerm@(overloadingHead(overloadingChildren))
+    vs.
+    inj{S2,injTo}(overloadedVariable)
+  If there exists a (unique) maximum overloadedHead with its result sort S2'
+  within S2 such that
+    inj{S2',injTo}(overloadedHead(freshVars)) =
+        overloadingHead(inj2(freshVars))
+
+  Then it narrows overloadedVariable to inj{S2',S2}(overloadedHead(freshVars))
+  and reduces the initial problem to
+    overloadingTerm@(overloadingHead(overloadingChildren))
+   vs
+    overloadingHead(inj2(freshVars))
+
+  Note that most instances of inj above are distinct.
+-}
+unifyOverloadingVsOverloadedVariable
+    :: MonadSimplify unifier
+    => InternalVariable variable
+    => Symbol
+    -> TermLike variable
+    -> ElementVariable variable
+    -> Attribute.FreeVariables variable
+    -> Inj ()
+    -> UnifyOverloadingResult unifier variable
+unifyOverloadingVsOverloadedVariable
+    overloadingHead
+    overloadingTerm
+    overloadedVar
+    freeVars
```

Comment:
Why does this function need to be passed the `FreeVariables`? Why doesn't it compute them itself?